flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobMaster.java

```diff
@@ -1014,7 +1014,6 @@ public void notifyAllocationFailure(AllocationID allocationID, Exception cause)
 	//-- job starting and stopping  -----------------------------------------------------------------
 
 	private Acknowledge startJobExecution(JobMasterId newJobMasterId) throws Exception {
-		validateRunsInMainThread();
```

Comment:
Why do we need to remove this call here?

Reply:
I removed it because the check failed because the man thread executor is only set after receiveing the fencing token, in `startJobMasterServices()`. I think now it would no longer cause problems as we have the dummy executor in place, however I think there is limited value anyways because, again, we are checking right at the start of `resetAndScheduleExecutionGraph`, the relevat method vere because it interacts with the EG.